tests/PHPStan/Reflection/PhpDefect/PhpDefectClassReflectionExtensionTest.php

```diff
@@ -8,178 +8,87 @@
 class PhpDefectClassReflectionExtensionTest extends \PHPStan\TestCase
 {
 
-	public function dataDomDocumentProperties(): array
-	{
-		return [
-			[
-				'documentURI',
-				'string',
-			],
-			[
-				'encoding',
-				'string',
-			],
-			[
-				'formatOutput',
-				'bool',
-			],
-			[
-				'implementation',
-				'bool',
-			],
-			[
-				'preserveWhiteSpace',
-				'bool',
-			],
-			[
-				'recover',
-				'bool',
-			],
-			[
-				'resolveExternals',
-				'bool',
-			],
-			[
-				'standalone',
-				'bool',
-			],
-			[
-				'strictErrorChecking',
-				'bool',
-			],
-			[
-				'substituteEntities',
-				'bool',
-			],
-			[
-				'validateOnParse',
-				'bool',
-			],
-			[
-				'version',
-				'string',
-			],
-			[
-				'xmlEncoding',
-				'string',
-			],
-			[
-				'xmlVersion',
-				'string',
-			],
-		];
-	}
-
 	/**
+	 * @dataProvider dataDateIntervalProperties
 	 * @dataProvider dataDomDocumentProperties
-	 * @param string $propertyName
-	 * @param string $typeDescription
+	 * @dataProvider dataZipArchiveProperties
+	 *
+	 * @param string $className
+	 * @param array  $data
 	 */
-	public function testDomDocumentProperties(string $propertyName, string $typeDescription)
+	public function testProperties(string $className, array $data)
 	{
-		$broker = $this->getContainer()->getByType(Broker::class);
-		$zipArchiveReflection = $broker->getClass(\DOMDocument::class);
-		$this->assertTrue($zipArchiveReflection->hasProperty($propertyName));
-		$propertyReflection = $zipArchiveReflection->getProperty($propertyName);
-		$this->assertInstanceOf(PhpDefectPropertyReflection::class, $propertyReflection);
-		$this->assertSame($zipArchiveReflection, $propertyReflection->getDeclaringClass());
-		$this->assertSame($typeDescription, $propertyReflection->getType()->describe());
+		foreach ($data as $propertyName => $typeDescription) {
+			$broker = $this->getContainer()->getByType(Broker::class);
+			$zipArchiveReflection = $broker->getClass($className);
+			$this->assertTrue($zipArchiveReflection->hasProperty($propertyName));
+			$propertyReflection = $zipArchiveReflection->getProperty($propertyName);
+			$this->assertInstanceOf(PhpDefectPropertyReflection::class, $propertyReflection);
+			$this->assertSame($zipArchiveReflection, $propertyReflection->getDeclaringClass());
+			$this->assertSame($typeDescription, $propertyReflection->getType()->describe());
```

Comment:
Problem with this line is that when it fails, I have no idea which property is it. Add a third parameter - message with `sprintf('%s::$%s', $className, $propertyName)` and test how it looks.